protocol_grpc.go

```diff
@@ -508,16 +513,17 @@ func grpcExtractErrorFromTrailer(trailers http.Header) *connect.Error {
 		return nil
 	}
 
-	code, err := strconv.ParseUint(codeHeader, 10 /* base */, 32 /* bitsize */)
+	code64, err := strconv.ParseUint(codeHeader, 10 /* base */, 32 /* bitsize */)
 	if err != nil {
 		return connect.NewError(
 			connect.CodeInternal,
 			protocolError("invalid error code %q: %w", codeHeader, err),
 		)
 	}
-	if code == 0 {
+	if code64 == 0 {
 		return nil
 	}
+	code := uint32(code64)
```

Comment:
Why introduce new variable and rename previous one? Seems like you could just stick with `code` above and elide this.

If this is to avoid a "nolint" for the type conversion on line 536, I think it'd be better with the nolint comment TBH.